sensor/common/scan/scan.go

```diff
@@ -133,8 +142,15 @@ func (s *LocalScan) EnrichLocalImageInNamespace(ctx context.Context, centralClie
 		return nil, errors.Join(ErrNoLocalScanner, ErrEnrichNotStarted)
 	}
 
+	scanLimitSemaphore := s.scanSemaphore
+
+	// Delegated requests
+	if req.ID != "" && !env.DelegatedScanningDisabled.BooleanSetting() {
+		scanLimitSemaphore = s.adHocScanSemaphore
+	}
```

Comment:
This `env.DelegatedScanningDisabled.BooleanSetting()` check should not be necessary. Let me know if you see otherwise.

Reply:
Want to double check: we don't want the delegated scanning check because this ad-hoc throttling is regardless of it's delegated or not?

Reply:
More so the only place that can sets the req.ID (currently) is the handler that already [performs this check](https://github.com/stackrox/stackrox/blob/d20bca1e92582829ade0146772f232fb5a0b84ed/sensor/common/delegatedregistry/delegated_registry_handler.go#L64-L66).